src/GitTfs/Commands/Verify.cs

```diff
@@ -15,28 +15,30 @@ namespace GitTfs.Commands
     [Description("verify [options] [commitish]\n   ex: git-tfs verify\n       git-tfs verify 889ad74c162\n       git-tfs verify tfs/mybranch\n       git-tfs verify --all")]
     public class Verify : GitTfsCommand
     {
+        private readonly RemoteOptions _remoteOptions;
         private readonly Help _helper;
         private readonly Globals _globals;
         private readonly TreeVerifier _verifier;
 
-        public Verify(Globals globals, TreeVerifier verifier, Help helper)
+        public Verify(Globals globals, TreeVerifier verifier, Help helper, RemoteOptions remoteOptions)
         {
             _globals = globals;
             _verifier = verifier;
             _helper = helper;
+            _remoteOptions = remoteOptions;
         }
 
         public OptionSet OptionSet
         {
             get
             {
                 return new OptionSet()
-            {
+                {
                     { "ignore-path-case-mismatch", "Ignore the case mismatch in the path when comparing the files.",
                         v => IgnorePathCaseMismatch = v != null },
                     { "all", "Verify all the tfs remotes",
                         v => VerifyAllRemotes = v != null },
-            };
+                }.Merge(_remoteOptions.OptionSet);
```

Comment:
> This change brings all the RemoteOptions to the verify command line. Does that make sense?

While trusting my own judgement at the time, it is a bit hard now to remember all the details from (almost) two years ago (and I obviously didn't find it interesting enough to further support/justify inside the commit message, eh)... :P

But I think the main point is that `verify` internally uses `GitTfsRemote`, and that one uses (and expects) `RemoteOptions`, so there is sense in being able to explicitly provide/use those here, too.

Please do note that this change (being commit https://github.com/git-tfs/git-tfs/commit/de0c7ea85054b2430cdec0192db871721df5477b) only triggers `GetAllOptions()` (inside `Ext.cs`) to recognize and parse `RemoteOptions` related command line options, too - it doesn't introduce any new logic in using/handling those options, as that already exists inside `verify` command through internal `GitTfsRemote` usage (just that user previously couldn't set them explicitly, through `verify` command line, and now it can).

> Shouldn't we just have created a GitIgnoreOption used wherever we need it?

If you really do mean standalone option related for .gitignore file handling only, then no. As described in commit (https://github.com/git-tfs/git-tfs/commit/1a60ec472b36a98821c1bedd5551679fe3cf741d) which moves `--gitignore` from `InitOptions` to `RemoteOptions`, it is tightly coupled to `--ignore-regex` and `--except-regex` options, already being found inside `RemoteOptions`, too (and all together used in `GitTfsRemote.IsIgnored()` path evaluation, for example).

But if you meant possibly to extract _all_ of these path ignoring options instead (into something like `IgnoreOption`) and reuse from there, than the answer is "maybe" - though that would probably be an orthogonal change, and out of scope of this series (and even then we might still end up using all of the current `RemoteOptions` options here, for `GitTfsRemote` sake).